scala/lambdas/entity-event-generator-lambda/src/main/scala/uk/gov/nationalarchives/entityeventgenerator/Lambda.scala

```diff
@@ -62,41 +58,46 @@ class Lambda extends LambdaRunner[ScheduledEvent, Int, Config, Dependencies] {
       entitiesClient: EntityClient[IO, Fs2Streams[IO]],
       dADynamoDBClient: DADynamoDBClient[IO],
       dASnsDBClient: DASNSClient[IO],
-      startFrom: Int,
       eventTriggeredDatetime: OffsetDateTime
-  ): IO[Int] =
+  ): IO[Int] = {
+    val ignoredEventTypes = List("Download", "Characterise", "VirusCheck")
     for {
       updatedSinceResponses <- dADynamoDBClient.getItems[GetItemsResponse, PartitionKey](
         List(PartitionKey("LastPolled")),
         config.lastEventActionTableName
       )
       updatedSinceResponse = updatedSinceResponses.head
       updatedSinceAsDate = OffsetDateTime.parse(updatedSinceResponse.datetime).toZonedDateTime
-      recentlyUpdatedEntities <- entitiesClient.entitiesUpdatedSince(updatedSinceAsDate, startFrom)
+      currentStart = updatedSinceResponse.start
+      recentlyUpdatedEntities <- entitiesClient.entitiesUpdatedSince(updatedSinceAsDate, currentStart)
       _ <- logger.info(s"There were ${recentlyUpdatedEntities.length} entities updated since $updatedSinceAsDate")
 
       entityLastEventActionDate <-
         if (recentlyUpdatedEntities.nonEmpty) {
           val lastUpdatedEntity: Entity = recentlyUpdatedEntities.last
           entitiesClient.entityEventActions(lastUpdatedEntity).map { entityEventActions =>
-            Some(entityEventActions.head.dateOfEvent.toOffsetDateTime)
+            Some(entityEventActions.filterNot(ev => ignoredEventTypes.contains(ev.eventType)).head.dateOfEvent.toOffsetDateTime)
```

Comment:
Is it possible that the last entity only has one of the ignored event types? if so, this would not generate correct date.